secure-staff-services.md

```diff
@@ -0,0 +1,69 @@
+- Log requests to all our services so that they can be centrally audited
+- Authenticate external requests before they reach our applications
+- Authenticate requests that come from within our network
+- Handle authorization and access management to restrict access to specific users
+
+## Recommendation
+
+This solution brings together multiple components to provide a single secure solution:
+
+### Authentication at the ALB
+
+For the case where an application is served via an [Application Load Balancer (ALB)](https://docs.aws.amazon.com/elasticloadbalancing/latest/application/introduction.html) in AWS we will use [authentication at the ALB](https://docs.aws.amazon.com/elasticloadbalancing/latest/application/listener-authenticate-users.html) with Google as an OpenID Connect (OIDC) compliant Identity Provider (IdP). This will require a Google OAuth client to be configured for your application as part of a Google Cloud project. 
+
+This approach allows us to authenticate external requests before they reach our applications.
+
+A proposed extension to this would be this [Cognito authentication proposal](https://docs.google.com/document/d/1bEyL_Hn7DRs7XNyhvjvDK0PGXRzxgFzCbMcnaVyXbA0/edit#heading=h.nchxv8vlw2w5) that provides further custom authentication with Google as an IdP but allows potential access management based on Google groups, and enforces MFA requirements. 
+
+This will remove the requirement to perform authentication within application code and provide a standard way of authenticating all our applications that can easily be [encoded in a GuCDK pattern](https://docs.google.com/document/d/1SfvjNRIzv1bNYRho7s5i_YZv6YyWadp4j9rNOqvpnJg/edit#heading=h.812bucjdpfq7).
+
+As a baseline services should require Guardian Google organisation membership with 2FA enabled, with the ability to perform tighter access management to a subset of Guardian staff.
+
+### Web Application Firewall (WAF)
+
+To provide a standard way to recognise and prevent common attacks than can come via web requests we will use AWS WAF [Web Access Control Lists (ACLs)](https://docs.aws.amazon.com/waf/latest/developerguide/web-acl.html). These can be integrated with various AWS solutions including AWS ALBs. See the following [Architecture Decision Record](https://github.com/guardian/waf/blob/main/adr/waf-management.md) for further context.
+
+This solution allows centrally managing Web ACL rules so that they are easily understood, contain an audit trail via source control and can easily be reviewed by InfoSec. In addition this solution can provide a [stream of request events](https://github.com/guardian/waf/blob/main/lib/primary-waf.ts#L62) that can be centrally audited.
+
+### Authentication at the application
+
+In order to ensure that requests that reach services (EC2 instances or Lambdas) inside our networks we should re-authenticate requests either by validating any tokens generated at the ALB or entirely re-authenticating the user.
```

Comment:
This sentence is a little hard to read; I can't suggest an edit, as I don't quite understand the sentiment 😬.

Reply:
I've tried to make this section clearer.